src/sort.rs

```diff
@@ -0,0 +1,131 @@
+//! # Sort
+//!
+//! Contains Sorter and SortOrder that use command line arguments
+
+// Only because of SortOrder, as a result of Clap bug/restriction that only permits bare variants
+// inside of arg!_enum macro call
+#![allow(missing_docs)]
+
+use core::cmp::Ordering;
+use fs_extra::dir::get_size;
+use std::ffi::OsStr;
+use std::path::PathBuf;
+use std::time::SystemTime;
+
+/// A structure that takes the SortOrder from command line arguments
+/// and reverse and sorts pathes based on original criteria
+// Chosen over functions, due to it being more adaptable in the future
+pub struct Sorter {
+    /// Order to sort the files
+    sort_order: SortOrder,
+    /// Whether or not to reverse the order of sorting
+    reverse: bool,
+    /// Maximum number of images to keep, remove the rest. 0 Meaning infinite
+    max_length: usize,
+}
+
+impl Sorter {
+    pub fn new(sort_order: SortOrder, reverse: bool, max_length: usize) -> Self {
+        Self {
+            sort_order,
+            reverse,
+            max_length,
+        }
+    }
+
+    /// Change the order of sorting, this function doesn't sort upon change
+    pub fn set_order(&mut self, sort_order: SortOrder) {
+        self.sort_order = sort_order;
+    }
+
+    /// Change whether or not to reverse the sorting of images
+    pub fn set_reverse(&mut self, reverse: bool) {
+        self.reverse = reverse;
+    }
+
+    /// Change the maximum length of the images vector, post sorting
+    pub fn set_max_length(&mut self, max_length: usize) {
+        self.max_length = max_length;
+    }
+
+    /// Sorts the images based on sort_order, reverses if necessary, and truncates as well
+    pub fn sort(&self, pathes: &mut Vec<PathBuf>) {
+        pathes.sort_by(|a, b| self.sort_order.file_compare(&a, &b));
+        if self.reverse {
+            pathes.reverse();
+        }
+        if self.max_length > 0 {
+            let new_length = pathes.len().saturating_sub(pathes.len() - self.max_length);
+            pathes.truncate(new_length);
```

Comment:
An issue that may need addressed later is needing to rescan all the paths if the user wants to reverse the order at runtime.

Addressing that would be more complicated in needing to keep around all the other images in `images`, but capping every other option which concerns amount of images at `max_length` amount of them.

Reply:
> An issue that may need addressed later is needing to rescan all the paths if the user wants to reverse the order at runtime.
> 
True, it would be more versatile to separate these, or at least include reverse as it's own function.

In that case would it be good to move `max_length` from `Sorter` to `Path` and just have it be an optional maximum index `Option<usize>`? That way we have an artificial cap, the user just won't see the images past `max_length` and that allows us to sort them later on.

Inexperience question:
If I were to commit these on my branch would I have to open up a new pull request or would the pull request be updated?

Reply:
> True, it would be more versatile to separate these, or at least include reverse as it's own function.

I thought of that.  The below code seems to not require adding your own, but reusing `images` reverse().

```rust
    pub fn sort<'a>(&self, pathes: &'a mut Vec<PathBuf>) -> &'a mut Vec<PathBuf> {
        pathes.sort_by(|a, b| self.sort_order.file_compare(&a, &b));
        if self.reverse {
            pathes.reverse();
        }
        if self.max_length > 0 {
            let new_length = pathes.len().saturating_sub(pathes.len() - self.max_length);
            pathes.truncate(new_length);
        }
        pathes
    }
```

So then you can do.

```rust
        sorter.sort(&mut images).reverse().truncate(22);
```


I'd definitely go with the optional `max_length` on Paths.  A different name such as `max_viewable` perhaps.

> If I were to commit these on my branch would I have to open up a new pull request or would the pull request be updated?

No. Pushing to the branch of this pull request will update this pull request.

Reply:
That looks like a great ease of use improvement onto sort, implementing that and the paths update now.